lnwallet/channel.go

```diff
@@ -1328,12 +1345,44 @@ func (lc *LightningChannel) fetchCommitmentView(remoteChain bool,
 		numHTLCs++
 	}
 
+	// Initiate feePerKw to the last committed fee for this chain.
+	feePerKw := commitChain.tail().feePerKw
+
+	// Check if any fee updates have taken place since that last commitment.
+	if lc.channelState.IsInitiator {
```

Comment:
Yeah they really should be. Although we'll need to add an update index to the update fee messages as well. With that, they can transparently be handled as any other update. Otherwise, without the index, we'll run into the same special handling case as we do currently. 

There might be some hacky thing we can do with the log counters on our side in order to emulate the existence of a log index for the update messages though.